crates/cheatcodes/src/inspector.rs

```diff
@@ -154,13 +153,40 @@ pub trait CheatcodesExecutor {
         None
     }
 
-    fn trace_zksync(
-        &mut self,
-        ccx_state: &mut Cheatcodes,
-        ecx: &mut Ecx,
-        call_traces: Vec<Call>,
-    ) {
-        self.get_inspector(ccx_state).trace_zksync(ecx, call_traces);
+    fn trace_zksync(&mut self, ccx_state: &mut Cheatcodes, ecx: Ecx, call_traces: Vec<Call>) {
+        let mut inspector = self.get_inspector(ccx_state);
+
+        // We recreate the EvmContext here to satisfy the lifetime parameters as 'static, with
```

Comment:
I don't quite understand this - where does the static bound show up?

Reply:
The `'static` is probably a misnomer, should've written it as 
> We recreate the EvmContext here to enforce the `Ecx` lifetime is completely contained within that of the inspector's as the compiler can no longer infer it.

In the code here https://github.com/matter-labs/foundry-zksync/blob/5e12e76ffbd55c777bd255ada1e5622dcd01a416/crates/cheatcodes/src/inspector.rs#L162 we use `&mut self` and `&mut ccx_state` to obtain the inspector which returns `Box<dyn InspectorExt + 'a>`, this does not play well with the compiler when we perform `trace_zksync(ecx)` which is now defined as `Ecx<'a, 'b, 'c> = &'a mut EvmContext<&'b mut (dyn DatabaseExt + 'c)>` so the compiler assumptions are broken as `&mut T` is invariant over `T`, causing us to explicitly define the lifetimes and their relationships as done in this commit [here ](https://github.com/matter-labs/foundry-zksync/blob/5e12e76ffbd55c777bd255ada1e5622dcd01a416/crates/cheatcodes/src/inspector.rs#L156)

The problem being, that we'd need to bubble this up several layers right upto the revm hook which we cannot update, in addition to being nasty.

That's why we redefine `&mut Ecx` by recreating it here.